src/qtt/measurements/acquisition/uhfli_scope_reader.py

```diff
@@ -0,0 +1,318 @@
+""" Provides oscilloscope functionality for the Zurich Instruments UHFLI."""
+
+import time
+from typing import List, Tuple, Optional
+
+import numpy as np
+from qcodes.instrument_drivers.ZI import ZIUHFLI
+from qilib.configuration_helper import InstrumentAdapterFactory
+from qilib.data_set import DataArray
+from qilib.utils import PythonJsonStructure
+
+from qtt.measurements.acquisition.interfaces import AcquisitionScopeInterface
+
+
+class UhfliScopeReader(AcquisitionScopeInterface):
+    """ Represents an acquisition wrapper for acquiring data with the Zurich Instruments UHFLI."""
+
+    def __init__(self, address):
+        """ Creates the connection to the UHFLI lock-in amplifier.
+
+        Args:
+            address: The unique address of the UHFLI.
+        """
+        super().__init__(address)
+        self.adapter = InstrumentAdapterFactory.get_instrument_adapter('ZIUHFLIInstrumentAdapter', address)
+        self.__uhfli = self.adapter.instrument
+        self.__acquisition_counter = 0
+
+    def initialize(self, configuration: PythonJsonStructure) -> None:
+        """ Applies the configuration to the UHFLI.
+
+        Args:
+            configuration: The configuration with all the UHFLI settings.
+        """
+        self.adapter.apply(configuration)
+
+    def start_acquisition(self) -> None:
+        """ Starts the acquisition mode of the scope in the UHFLI."""
+        self.__uhfli.scope.set('scopeModule/mode', 1)
+        self.__uhfli.scope.subscribe('/{0}/scopes/0/wave'.format(self._address))
+        self.__uhfli.daq.sync()
+        self.__uhfli.daq.setInt('/{0}/scopes/0/enable'.format(self._address), 1)
+
+    def acquire(self, number_of_records: int=1, timeout: float=30) -> List[DataArray]:
+        """ Collects records from the UHFLI.
+
+        Args:
+            number_of_records: The number of records which should be collected at once.
+            timeout: The time the collecting of records can maximally take before raising an error.
+
+        Returns:
+            A list with the recorded scope records as data arrays.
+        """
+        collected_records = [self.__create_setpoint_data()]
```

Comment:
Could also attach the setpoint array to the data array instead:
`DataArray(name="...", label="...", unit="...", set_arrays=[self.__create_setpoint_data()])`